JENKINS.md

```diff
@@ -0,0 +1,123 @@
+
+## Job Creation
+* Go to Jenkins Home Page -> New Item
+* Provide a *Item Name* (Such as "cordova-plugin-console")
+* Choose *Multi-configuration project*
+* Click *Ok*
+* Provide a *Description*
+* Check *GitHub Project*. Provide the correct project URL. (__Note__: This is an important parameter. GitHub Pull Request will not work without this value)
+* Check *This build is parameterized*. 
+    * Add a String Parameter
+    * Add *ghprbActualCommit* as *Name*
+    * Add *master* as *Default Value*
+    * This value will be used to start parameterized builds manually. 
+* Under *Source Code Management*, Choose *Multiple SCMs*
+* Add each repository you want to use. Normally for paramedic testing (say for cordova-plugin-console), you need 3 repositories
+    * cordova-plugin-console
+    * cordova-paramedic
+    * cordova-medic (for configuration files)
+* For the specific repository that is tested, specify **${ghprbActualCommit}** as the branch to build. For other two repositories, you could use master.
+* Checkout each repository to a sub-directory. (This option could be found under Additional Behaviors)
+* Under *Build Trigger*, Choose *GitHub Pull Request Builder*. 
+* Under *Advanced*:
+    * Choose a Trigger phrase
+    * Choose the correct crontab
+    * Choose *Build every pull request automatically without asking (Dangerous!).*
+* Under *Trigger Setup*, the following values need to be configured:
+    * *Commit Status Build Triggered*: --none--
+    * *Commit Status Build Started*: --none--
+    * *Commit Status Build Result*: Provide appropriate messages for success and failure
+* Under *Configuration Matrix*:
+    * Add *Slaves* axis
+    * Add *User Defined axis*
+        * Provide platformName as Name
+        * Provide the following values
+            * windows-8.1-store
+            * windows-10-store
+            * windows-8.1-phone
+            * ios
+            * android
+    * Check *Combination Filter* with the following filter:
+    
+```(label == "mac-slave" && platformName == "ios") || (label == "mac-slave" && platformName == "android") || (label == "windows-slave" && platformName == "windows-10-store") || (label == "windows-slave" && platformName == "windows-8.1-phone") || (label == "windows-slave" && platformName == "windows-8.1-store") || (label == "windows-slave" && platformName == "android")```
+* Under *Build*:
+    * Add *Execute Shell* step with the following commands:
+        * ``cd cordova-paramedic``
+        * ``npm install``
+    * Add a second *Execute Shell* step with the following commands:
+        * ```node cordova-paramedic/main.js --config $WORKSPACE/cordova-medic/jenkins-conf/$platformName.config.json --plugin $WORKSPACE/cordova-plugin-console --outputDir $WORKSPACE --tccDb $WORKSPACE/cordova-medic/lib/tcc/TCC.db```
```

Comment:
My concern is that we will continue to change this and it will be very difficult to maintain this document. Is there a way to export the Jenkins configuration?